lightning/src/ln/channel.rs

```diff
@@ -3030,6 +3043,7 @@ impl<SP: Deref> Channel<SP> where
 		};
 
 		self.context.cur_holder_commitment_transaction_number -= 1;
+		self.context.expecting_peer_commitment_signed = false;
```

Comment:
Just noting: I was thinking this should maybe be set to false only after the monitor update has persisted, but it seems fine since this is not a problem of having the latest commitment (we're coop closing anyway) and more of updating the latest channel state to be able to properly coop close (and also this PR adds a check that a monitor update is not in progress before sending closing_signed so technically that covers this anyway)